sdk/containerregistry/azure-containerregistry/azure/containerregistry/_container_registry_client.py

```diff
@@ -875,42 +886,92 @@ def download_manifest(self, repository, tag_or_digest, **kwargs):
-        :returns: DownloadBlobResult or None
-        :rtype: ~azure.containerregistry.DownloadBlobResult or None
+        :returns: DownloadBlobResult
+        :rtype: ~azure.containerregistry.DownloadBlobResult
         :raises ValueError: If the parameter repository or digest is None.
         """
-        try:
-            deserialized = self._client.container_registry_blob.get_blob( # type: ignore
-                repository, digest, cls=_return_deserialized, **kwargs
-            )
-        except ValueError:
-            if repository is None or digest is None:
-                raise ValueError("The parameter repository and digest cannot be None.")
-            raise
 
-        if deserialized:
-            blob_content = b''
-            for chunk in deserialized:
-                if chunk:
-                    blob_content += chunk
-            return DownloadBlobResult(data=BytesIO(blob_content), digest=digest)
-        return None
+    @overload
+    def download_blob(self, repository: str, digest: str, destination: str, **kwargs) -> None:
+        """Download a blob that is part of an artifact to a file in destination.
+
+        :param str repository: Name of the repository.
+        :param str digest: The digest of the blob to download.
+        :param str destination: The destination of the blob download to.
+        :returns: None
+        :raises ValueError: If the parameter repository or digest is None.
+        """
+
+    @distributed_trace
+    def download_blob(self, *args: str, **kwargs) -> Union[DownloadBlobResult, None]:
+        repository = args[0]
+        digest = args[1]
+        if len(args) == 2:
+            try:
+                deserialized = self._client.container_registry_blob.get_blob(
+                    repository, digest, **kwargs
+                )
+                blob_content = b''
+                for chunk in deserialized:
+                    if chunk:
+                        blob_content += chunk
+                blob_stream = BytesIO(blob_content)
+                if not _validate_digest(blob_stream, digest):
+                    raise ValueError("The requested digest does not match the digest of the received blob.")
+            except ValueError:
+                if repository is None or digest is None:
+                    raise ValueError("The parameter repository and digest cannot be None.")
+                raise
+            return DownloadBlobResult(data=blob_stream, digest=digest)
+        else:
+            destination = args[2]
+            self._download_blob_chunk(repository, digest, destination, **kwargs)
+    
+    def _download_blob_chunk(self, repository: str, digest: str, destination: str, **kwargs) -> None:
+        import datetime, os
+        file_name = datetime.datetime.now().strftime("%Y-%m-%d%H%M%S")
+        downloaded = 0
+        blob_size = None
+        hasher = hashlib.sha256()
+        with open(os.path.join(destination, file_name), 'wb') as file:
```

Comment:
As discussed, we shouldn't need to create the file inside this method.

Reply:
Updated to download to a stream